src/Symfony/Bundle/TwigBundle/Cache/ChainCache.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\TwigBundle\Cache;
+
+use Twig\Cache\CacheInterface;
+
+/**
+ * Chains several caches together.
+ *
+ * Cached items are fetched from the first cache having them in its data store.
+ * They are saved and deleted in all adapters at once.
+ */
+class ChainCache implements CacheInterface
+{
+    /**
+     * @param CacheInterface[] $caches The ordered list of caches used to store and fetch cached items
+     */
+    public function __construct(
+        private array $caches,
+    ) {
+        if (!$caches) {
+            throw new \InvalidArgumentException('At least one cache must be specified.');
+        }
+
+        foreach ($caches as $cache) {
+            if (!$cache instanceof CacheInterface) {
+                throw new \InvalidArgumentException(sprintf('The class "%s" does not implement the "%s" interface.', get_debug_type($cache), CacheInterface::class));
+            }
+        }
+    }
+
+    public function generateKey(string $name, string $className): string
+    {
+        return $className.'#'.$name;
+    }
+
+    /**
+     * @return array{name: string, className: string}
+     */
+    protected function splitKey(string $key): array
+    {
+        [$className, $name] = explode('#', $key, 2);
+
+        return ['name' => $name, 'className' => $className];
+    }
+
+    public function write(string $key, string $content): void
+    {
+        ['name' => $name, 'className' => $className] = $this->splitKey($key);
+
+        foreach ($this->caches as $cache) {
+            $cache->write($cache->generateKey($name, $className), $content);
+        }
```

Comment:
This can be more memory efficient if `splitKey` return a list, and is used to call directly `$cache->generateKey`
```suggestion
        $key = $this->splitKey($key);

        foreach ($this->caches as $cache) {
            $cache->write($cache->generateKey(...$key), $content);
        }
```